### DISCUSSION TITLE
GLTF not loaded properly

### DISCUSSION THREAD

**Discussion by Efor18** [sceneview/sceneform-android#103] [Q&A]
Hi, 

I'm trying to load this GlTF model but is not loaded properly. It shows some black lines and is not showing the car.

The model can be downloaded from here:

https://sketchfab.com/3d-models/free-honda-e-detailed-interior-2021-814cf3f8cfd64594b1c4f96e873c3cef

Any idea why? Is loading perfectly in iOS

Thanks in advance

**Comment by ThomasGorisse** [comment]
I have no issue with your model:
![image](https://user-images.githubusercontent.com/6597529/123125873-3281ca80-d449-11eb-91e6-cd380510ecbe.png)
![image](https://user-images.githubusercontent.com/6597529/123126482-b5a32080-d449-11eb-8739-9be3ee740a8a.png)

Make sure to compress it with draco since the original file size is 200Mo and to scale it down a lot (0.01)

**Reply by Efor18** [reply]
Oh wow!! Thanks @ThomasGorisse!

Can you help me with this compression, I'm new to this world and I'm a bit lost, so I compress it the gltf files with the following command:

`gltf-pipeline -d -i scene.gltf -o scene.glb`

But the glb resulting file is still very big: 134 MB and the app crash when I try to load it.

Is there any other way of compress it?

Regarding the scale it down I guess is just add something like this to the code:

```
val model = TransformableNode(arFragment.transformationSystem)
        model.setParent(anchorNode)
        model.renderable = this.model
        model.scaleController.maxScale = 0.01f
        model.scaleController.minScale = 0.001f
        model.select()
```

Am I right?

Thank you very much for your help!!

**Comment by ThomasGorisse** [comment]
Blender is your friend for that.
Import your .gltf inside Blender.
Remove the defaults entities (Camera, Light, Cube)
Scale down your model to 0.01
Export to glb with compression checked.

BUT 124 MB is a little bit ambitious for a 3D asset mobile AR usage.
For reducing a 3D model size, you have multiple choices:

Low poly it (= less precise rendering) by reducing the faces and vertices count.
Check that you use as less textures as possible and replace them with material extensions usage.
Delete some not accurate parts of your model.
In your case, the last choice may be the best since the car has a lot of details inside herself which you could not need to make visible to users.

With those optimizations I was able to reduce some 3D topographic drone datas from 1 GB to 25 MB and still make the monument appear at a 1:1 geographic scale within Sceneform without out of memory and without lost details.

Blender is awesome after you spend a little time to find where things are

**Reply by Efor18** [reply]
Thank you very much!!
